include/onmt/BPE.h

```diff
@@ -16,17 +17,33 @@ namespace onmt
 
     std::vector<std::string> encode(const std::string& str) const override;
 
+    void init_bpe_vocab(std::string path, int bpe_vocab_threshold);
+    void set_joiner(std::string joiner)
```

Comment:
What about spacers?

Would it make sense to pass the joiner and spacer as a list to `init_bpe_vocab` which will remove such substrings in each word when creating `_bpe_vocab`?